### PR TITLE
Add JSON custom error types. Refactoring.

### PR DESCRIPTION
**Pull request by sept-en**
* Add `JsonError` for `json` module.
* Use `JsonError` instead `()` for errors in `JsonValue`.
* Don't panic in `json` module on parsing.
* Move `SdkUnwrap`, `SdkExpect` traits from `lib.rs` to `types.rs`.
* Unwrap the error when parsing JSON for parameter args instead of failing with generic `ERR_FAILED_PARSE`.

### CODE REVIEW

**Review by birchmd — APPROVED** [review]
LGTM! Nice job!

File: src/types.rs
Comment (joshuajbouw) [inline src/types.rs:238]: To be fair, this should be somewhere else. I wouldn't exactly call them types.

**Review by joshuajbouw — CHANGES_REQUESTED** [review]

File: src/types.rs
Comment (sept-en) [reply src/types.rs:238]: Perhaps, yes, I agree!. Please also note we have similar traits in the same `types` file.
So if you don't mind, I suggest merging this PR and leave this part for the next one with some refactoring and structuring changes.

Comment (birchmd) [reply src/types.rs:238]: I agree with @sept-en that we should not block this PR over this sort of code organization. If it really is `P0` as the tag on the PR says then getting through and fixing minor comments later seems better than bikeshedding on it now.

Comment (sept-en) [reply src/types.rs:238]: @joshuajbouw @birchmd created issue #134 for that. Please, let's merge this PR ASAP if possible :)

**Review by joshuajbouw — APPROVED** [review]
Given comments, yes we can my issue to another PR.

**Review by mrLSD — APPROVED** [review]

**Review by artob — APPROVED** [review]

**Pull request merged**
